STYLE.md

```diff
@@ -0,0 +1,318 @@
+
+-	Subshell is being run to fill a variable's value.
+
+-	Variable is being expanded.
+
+When expanding variables inside another variables put the variable being
+expanded inside brackets and quote the whole variable.
+
+**pkgname**
+and
+**version**
+must never be quoted, if they need to be quoted then they don't satisfy the
+rules for those fields
+
+If using variables that are not part of xbps-src as defined per the manual then
+prefix them with a underscore.
+
+## CONTROL STRUCTURES
+
+The variables are over, now begins the
+**control structures**
+which are any conditionals and structures provided by the shell to control the flow
+of execution of the template, like:
+
+-	if EXPRESSION; then COMMAND-LIST; fi
+
+-	case EXPRESSION in CASE1) COMMAND-LIST;; ... CASEN) COMMAND-LIST;; esac
+
+-	for NAME \[in LIST ]; do COMMANDS; done
+
+-	while CONTROL-COMMAND; do CONSEQUENT-COMMANDS; done
+
+they should start 1 blank line after the variables, and between each occurrence
+of a
+**control structure**
+there should be 1 blank line of separation, there is no specific ordering for
+control structures.
+
+When declaring one make the start of the declaration in one line,
+1 hard tab of indentation applies inside the control structure.
+
+When using a control structure there should be 1 whitespace separating the
+semicolon and the right-side keyword.
+No spacing is needed between the semicolon and the left side of the declaration.
+The whole declaration must be on one line.
+
+when checking if a variable is not empty via
+**\[ ]**
+the use of -n can be discarded.
+
+	# Wrong!
+	# space between ] and ;
+	if [ -n "$CROSS_BUILD" ] ; then
+		hostmakedepends+=" glib-devel"
+	fi
+	
+	# Wrong!
+	# then keyword should be on the same line
+	if [ -n "$CROSS_BUILD" ]
+	then
+		hostmakedepends+=" glib-devel"
+	fi
+	
+	# Correct!
+	if [ "$CROSS_BUILD" ]; then
+		hostmakedepends+=" glib-devel"
+	fi
+
+## BUILD PHASES
+
+After control structures comes the
+**build phases**
+which are functions.
+That means they have 1 hard tab of indentation inside them.
+They also follow the same spacing rules as
+control structures
+and their ordering should follow what is run on when building the package, per
+example:
+**do\_install**
+comes after
+**post\_build**.
+
+Inside
+**build phases**
+there is no strict style for separation between variables, control structures and
+any other content, but it should be done if it helps to make it readable.
+
+## SUBPACKAGES
+
+The last section for the template is the
+**subpackages**
+declarations which are functions that are marked
+**&lt;subpkgname&gt;\_package()**,
+the same spacing and indentation rules of
+**build phases**
+apply here.
+
+## EXTRA GUIDELINES
+
+The following are guidelines that apply to the whole structure of the template and
+guidelines for when following this document.
+
+Don't be smart when writing a template, keep it clear and readable.
+Other people and future you appreciate.
```

Comment:
Substitution of constant like ${pkgname}, ${homepage} (rarely changes) etc. has little value if any, makes less readable and makes harder to work with (e.g. void-updates.txt has unprocessed substitutions). Can it be stated explicitly that variable substitition of constants is discouraged?

I would also apply that to ${pkgname#perl-} and similar constants that may look reasonable for reuse for other template, but two occurrences of name need to be search&replaced anyway - in pkgname= and header - so another occurrence makes no difference.

https://github.com/void-linux/void-packages/pull/94/files#r221455751